.github/workflows/docs.yml

```diff
@@ -1,50 +1,80 @@
-name: Build Docs
+name: Build Documentation
 
-# We don't want pushes (or PRs) to gh-pages to kick anything off
 on:
+  workflow_dispatch:
   push:
-    branches: [ main ]
+    branches: [main]
+    paths:
+      - tests/*
+      - hydromt_wflow/*
+      - docs/*
+      - examples/*
+      - pyproject.toml
   pull_request:
-    branches: [ main ]
+    branches: [main]
+    paths:
+      - tests/*
+      - hydromt_wflow/*
+      - docs/*
+      - examples/*
+      - pyproject.toml
 
 jobs:
   # Build docs on Linux
   Docs:
-    name: linux docs
-    runs-on: ubuntu-latest
-    env:
-      DOC_VERSION: dev
     defaults:
       run:
         shell: bash -l {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest" ] #, "macos-latest", "windows-latest"]
+        python-version: ['3.11']
+        include:
+        - os: ubuntu-latest
+          label: linux-64
+          prefix: /usr/share/miniconda3/envs/hydromt_wflow
+
+    name: ${{ matrix.label }}
+    runs-on: ${{ matrix.os }}
+
     steps:
       - name: Checkout source
         uses: actions/checkout@v3
-      - name: install tomli
-        run: pip install tomli
-      - name: Generate env spec
-        run: python make_env.py doc
-      - name: Setup environment
+
+      - name: Setup Mambaforge
         uses: conda-incubator/setup-miniconda@v2
         with:
-          python-version: "3.9"
+          python-version: ${{ matrix.python-version }}
           miniforge-variant: Mambaforge
-          channels: conda-forge,defaults
-          channel-priority: strict
-          environment-file: environment.yml
-          activate-environment: hydromt-wflow
+          miniforge-version: latest
+          activate-environment: hydromt_wflow
+          use-mamba: true
+
+      - name: Generate env spec
+        run: |
+          pip install tomli
+          python make_env.py doc --py-version ${{ matrix.python-version }}.*
+
+      - name: Update environment
+        run: mamba env update -f environment.yml
 
       - name: Install HydroMT-Wflow plugin
         run: pip install .
 
-      - name: Build docs
+      - name: Generate dummy docs
+        if: ${{ github.event_name == 'pull_request' }}
         run: |
-          pushd docs
-          make html
-          popd
+          export PATH=/usr/share/miniconda3/bin:$PATH
+          PYTHONPYCACHEPREFIX=~/pycache mamba run -n hydromt_wflow sphinx-build ./docs ./docs/_build -b dummy
```

Comment:
You don't seem to use the same caching strategy as core so the export and cache previx is not necessary here. if you want me to explain why this is necessary in core but not here ping me

Reply:
Will do. Will look at that.